lib/src/obsoleted/rules/always_remove_listener_rule.dart

```diff
@@ -0,0 +1,266 @@
+import 'package:analyzer/dart/analysis/results.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:collection/collection.dart';
+
+import '../../models/issue.dart';
+import '../../models/severity.dart';
+import '../../utils/node_utils.dart';
+import '../../utils/rule_utils.dart';
+import 'obsolete_rule.dart';
+
+class AlwaysRemoveListenerRule extends ObsoleteRule {
+  static const String ruleId = 'always-remove-listener';
+  static const _documentationUrl = 'https://git.io/JO7VM';
+
+  static const _warningMessage =
+      'Listener is not removed. This might lead to a memory leak.';
+
+  AlwaysRemoveListenerRule({Map<String, Object> config = const {}})
+      : super(
+          id: ruleId,
+          documentationUrl: Uri.parse(_documentationUrl),
+          severity: readSeverity(config, Severity.warning),
+          excludes: readExcludes(config),
+        );
+
+  @override
+  Iterable<Issue> check(ResolvedUnitResult source) {
+    final _visitor = _Visitor();
+
+    source.unit?.visitChildren(_visitor);
+
+    return _visitor.missingInvocations
+        .map((invocation) => createIssue(
+              rule: this,
+              location: nodeLocation(
+                node: invocation,
+                source: source,
+                withCommentOrMetadata: true,
```

Comment:
@incendial please remove this argument. (this argument useful for get function or method node location)

Reply:
Done.